Week1

elena-sek opened the pull request:
Created R package "ProductivityApp" which contains 3 functions so far:` create_empty_task_df, add_task, generate_eisenhower_matrix`. These functions also include descriptions.

elena-sek committed:
Created the first version of the R package

Created the R skeleton which currently has 3 functions. Updated the description of the ProductivityApp.

elena-sek committed:
Added the tar.gz file

elena-sek committed:
Added function descriptions

Added descriptions to the 3 functions that I have so far.

elena-sek committed:
Updated 3 functions and their descriptions

Deleted function create_empty_task_csv and created create_empty_task_df function instead. Updated add_task and generate_eisenhower_matrix functions and their descriptions.

jessiwitti commented:
In general the basic structure of the R package is there + there is documentation about every function. I really like your first functions, it seems straight forward and works fine. A small suggestion: Maybe you can add to your create empty task description that no argument is needed (and maybe even does not work with an argument). Good work! :)

brunykrijgsman commented on ProductivityApp/R/ProductivityApp_functions.R:
```diff
@@ -0,0 +1,84 @@
+}
+
+#' @title Generate an Eisenhower Matrix
+#' @description Creates an Eisenhower matrix based on a data frame with tasks.
```
Suggestion: Make it clearer what an Eisenhower matrix represents in the discription.

brunykrijgsman commented on ProductivityApp/R/ProductivityApp_functions.R:
```diff
@@ -0,0 +1,84 @@
+
+#' @title Add Task 
+#' @description Adds a new task with its variables to the task data frame with columns Task, Duration, Importance, Urgency, and Enjoyability.
+#' @param df name of the task data frame that will be updated
```
Suggestion: To make it more consistent, add the parameter type to this df parameter too.

brunykrijgsman reviewed (approved):
It's a super cool project! 
- Everything is included in the R package skeleton.
- The documentation of the functions is clear (I made some suggestions in the code).

akane7011 reviewed (commented):
Looks very clear and well-documented to me! I suppose this is for prioritizing some stuff over others, so maybe it would also be nice to color things that should be prioritized in the plot.

elena-sek committed:
Incorporated feedback

Updated documentation of the 3 original functions using \itemize and providing more details in the description of the generate_eisenhower_matrix function.
